raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -1165,15 +1179,26 @@ def handle_refundtransfer(
         if not is_valid:
             return TransitionResult(mediator_state, channel_events)
 
+        # We remove all routes where the sender of the refund is our forward hop
```

Comment:
This is what `filter_used_routes` does (or is suppose to do), please use that function.